src/Parser/Rogue/Assassination/Modules/Legendaries/DuskwalkersFootpads.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+
+import ITEMS from 'common/ITEMS';
+import SPELLS from 'common/SPELLS';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import SpellUsable from 'Parser/Core/Modules/SpellUsable';
+import RESOURCE_TYPES from 'common/RESOURCE_TYPES';
+import SpellLink from 'common/SpellLink';
+
+// Equip: The remaining cooldown on Vendetta is reduced by 1 sec for every 65 Energy you expend.
+
+const VENDETTA_CDR_PER_ENERGY = 1/65;
+
+class DuskwalkersFootpads extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+    spellUsable: SpellUsable,
+      };
+
+  totalReduction = 0;
+  wastedReduction = 0;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasFeet(ITEMS.DUSKWALKERS_FOOTPADS.id);
+  }
+
+  on_byPlayer_spendresource(event) {
+    if (event.resourceChangeType !== RESOURCE_TYPES.ENERGY.id) {
+      return;
+    }
+
+    const spent = event.resourceChange;
+
+    if (!this.spellUsable.isOnCooldown(SPELLS.VENDETTA.id)) {
+      this.wastedReduction += spent * VENDETTA_CDR_PER_ENERGY;
+    } else {
+
+    this.totalReduction += spent * VENDETTA_CDR_PER_ENERGY;
```

Comment:
In order for Vendetta's CD to keep on showing correctly in the cooldowns tracker, you should also inform spellUsable of this reduction using `this.spellUsable.reduceCooldown( ...`. Note that spell usable wants the reduction in milliseconds, so you'll need to multiply the reduction by 1000.

Reply:
copypasting from my comment that probably got buried when github collapsed all the requests: 
Looking more into this, I forgot something quite important: 
You will need to call ``this.spellUsable.reduceCooldown(SPELLS.VENDETTA.id, cooldownReducedAmount)`` to properly reduce the cooldown of the spell aswell. 

So probably ends up looking like (with whatever variable names to replace my temp ones)
```jsx 
const possibleCDR = spent * VENDETTA_CDR_PER_ENERGY*1000;
if(this.spellUsable.isOnCooldown(SPELLS.VENDETTA.id)) {
      const reductionMs = this.spellUsable.reduceCooldown(SPELLS.VENDETTA.id, possibleCDR);
      this.effectiveVendettaReductionMs += reductionMs;
      this.wastedVendettaReductionMs += (possibleCDR - reductionMs);
    } else {
      this.wastedVendettaReductionMs += possibleCDR;
}
```